### PR TITLE
1214 hash examples

### PR DESCRIPTION
**Pull request by Arithmeticus**
addresses #1214

### CODE REVIEW

**Review by ChristianGruen — APPROVED** [review]
Thanks.

File: specifications/xpath-functions-40/src/function-catalog.xml [hunk 1]
```diff
@@ -5238,17 +5231,16 @@ return normalize-unicode(concat($v1, $v2))</eg>
...
+               <fos:result>xs:hexBinary("00000000")</fos:result>
             </fos:test>
             <fos:test>
-               <fos:expression><eg>hash("input", { "algorithm": "CRC-32" }) => string()</eg></fos:expression>
-               <fos:result>"D82832D7"</fos:result>
+               <fos:expression><eg>hash("input", { "algorithm": "CRC-32" })</eg></fos:expression>
+               <fos:result>xs:hexBinary("D82832D7")</fos:result>
             </fos:test>
             <fos:test use="v-hash-salt">
                <fos:expression>hash("password123" || $salt, "sha-1234567")</fos:expression>
```
Comment (ChristianGruen) [inline specifications/xpath-functions-40/src/function-catalog.xml:5246]:
```suggestion
               <fos:expression>hash("password123" || $salt, { "algorithm": "sha-1234567" })</fos:expression>
```

Comment (ChristianGruen) [inline specifications/xpath-functions-40/src/function-catalog.xml:5246]: @Arithmeticus Could you fix the last example for me, for which I had removed the comment recently?

```suggestion
               <fos:expression>hash("password123" || $salt, { "algorithm": "sha-1234567" })</fos:expression>
```

**Comment by ndw** [comment]
The CG agreed to merge this PR without discussion at [meeting 078](https://qt4cg.org/meeting/minutes/2024/05-21.html).

**Pull request merged**